bftengine/src/bftengine/KeyManager.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "KVBCInterfaces.h"
+#include "Serializable.h"
+
+class KeyManager {
+ public:
+  struct KeyExchangeMsg : public concord::serialize::SerializableFactory<KeyExchangeMsg> {
+    std::string key_;
+    std::string signature_;
+    uint32_t repID_;
+    std::string toString() const;
+    static KeyExchangeMsg MsgDeserializer(const char* serStr, const uint32_t& size);
+
+   protected:
+    const std::string getVersion() const;
+    void serializeDataMembers(std::ostream& outStream) const;
+    void deserializeDataMembers(std::istream& inStream);
+  };
+
+ private:
+  concord::kvbc::IClient* cl_{nullptr};
+  int repID_{};
+
+ public:
+  std::atomic_bool keysExcahnged_{false};
+  static KeyManager& get();
```

Comment:
If we do like singletons, then ctor has to be private, copy/assign/move-copy/move-assign attributes to be deleted.